### PR TITLE
Entity Settings Page to MWC 3

### PR DESCRIPTION
**Pull request by zsarnett**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/entities/entity-registry-settings.ts [hunk 1]
```diff
@@ -137,15 +136,16 @@ export class EntityRegistrySettings extends SubscribeMixin(LitElement) {
             </div>
           `
         : ""}
-      ${this._error ? html` <div class="error">${this._error}</div> ` : ""}
+      ${this._error ? html`<div class="error">${this._error}</div>` : ""}
```
Comment (bramkragten) [inline src/panels/config/entities/entity-registry-settings.ts:139]:
```suggestion
      ${this._error ? html`<ha-alert alert-type="error">${this._error}</ha-alert>` : ""}
```

File: src/panels/config/helpers/forms/ha-input_number-form.ts [hunk 2]
```diff
@@ -181,15 +181,18 @@ class HaInputNumberForm extends LitElement {
     const target = ev.target as any;
     const configValue = target.configValue;
     const value =
-      target.type === "number" ? Number(ev.detail.value) : ev.detail.value;
+      target.type === "number"
+        ? Number(target.value)
+        : ev.detail.value || target.value;
```
Comment (bramkragten) [inline src/panels/config/helpers/forms/ha-input_number-form.ts:186]: `detail` is undefined on an input event I think?

Comment (zsarnett) [reply src/panels/config/helpers/forms/ha-input_number-form.ts:186]: Its for the icon picker

Comment (bramkragten) [reply src/panels/config/helpers/forms/ha-input_number-form.ts:186]: yes, but will throw of other inputs

Comment (zsarnett) [reply src/panels/config/helpers/forms/ha-input_number-form.ts:186]: Oh I see what you are saying.

File: src/panels/config/entities/entity-registry-basic-editor.ts [hunk 3]
```diff
@@ -123,16 +122,16 @@ export class HaEntityRegistryBasicEditor extends SubscribeMixin(LitElement) {
       computeDomain(this.entry.entity_id);
 
     return html`
-      <paper-input
+      <ha-textfield
+        error-message="Domain needs to stay the same"
```
Comment (balloob) [inline src/panels/config/entities/entity-registry-basic-editor.ts:126]: This is not a property for textfield https://github.com/material-components/material-web/tree/master/packages/textfield#propertiesattributes

File: src/panels/config/helpers/forms/ha-counter-form.ts [hunk 4]
```diff
@@ -68,10 +68,10 @@ class HaCounterForm extends LitElement {
...
       <div class="form">
-        <paper-input
+        <ha-textfield
           .value=${this._name}
           .configValue=${"name"}
-          @value-changed=${this._valueChanged}
+          @input=${this._valueChanged}
           .label=${this.hass!.localize(
             "ui.dialogs.helper_settings.generic.name"
           )}
```
Comment (balloob) [inline src/panels/config/helpers/forms/ha-counter-form.ts:77]: same on line below

File: src/panels/config/entities/entity-registry-settings.ts [hunk 5]
```diff
@@ -176,16 +179,16 @@ export class EntityRegistrySettings extends SubscribeMixin(LitElement) {
               )}
             </mwc-select>`
           : ""}
-        <paper-input
+        <ha-textfield
+          error-message="Domain needs to stay the same"
```
[the remark this answers is not in the document]
Comment (zsarnett) [reply src/panels/config/entities/entity-registry-settings.ts:183]: https://github.com/home-assistant/frontend/blob/dev/src/components/ha-textfield.ts#L10

**Review by balloob — APPROVED** [review]

**Pull request merged**